### PR TITLE
✨ Reduce dependency graph for API module

### PR DESCRIPTION
**Pull request by akutz**
**What does this PR do, and why is it needed?**

This patch updates the API module so its dependency graph is greatly reduced. The dependency on controller-runtime for things other than conversion was creating compile-time build issues for consumers of the API module.

The change encompasses three things:

1. Following the guide https://main.cluster-api.sigs.k8s.io/developer/providers/implementers-guide/create_api#registering-apis-in-the-scheme which demonstrates how to remove controller-runtime for registering API types with the scheme builder.

2. Moving the webhook registrations out of the API module. The module already did not have the validation/mutation hooks, but did, and still, has the conversion hooks since those *must* be in the API package. However, their hook registrations were moved out of the API module since the method of registration depended upon controller-runtime.

3. Removed the dependency on the testlabels module and just used the strings "api" and "fuzz" directly.

**Which issue(s) is/are addressed by this PR?** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

Fixes `NA`

**Are there any special notes for your reviewer**:

**Please add a release note if necessary**:

```release-note
Reduce dependency graph for API module.
```

### CODE REVIEW

**Review by dougm — APPROVED** [review]
lgtm

**Review by dilyar85 — APPROVED** [review]
LGTM!

File: webhooks/conversion/v1alpha1/webhooks.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2023-2024 VMware, Inc. All Rights Reserved.
```
Comment (aruneshpa) [inline webhooks/conversion/v1alpha1/webhooks.go:1]: I think this just needs to be 2024?  Same in other files.

**Review by aruneshpa — APPROVED** [review]
One nit related to copyright years.  LGTM otherwise.

File: webhooks/conversion/v1alpha1/webhooks.go (hunk 1 above)
Comment (akutz) [reply webhooks/conversion/v1alpha1/webhooks.go:1]: Ah, yeah, I moved them from the original location and just added the 2024. I only renamed them later. I'll update, thanks!

**Review by bryanv — APPROVED** [review]

**Pull request merged**